kore/test/Test/Kore/Builtin/Set.hs

```diff
@@ -313,6 +313,67 @@ test_unifyFramingVariable =
             (===) expect =<< evaluate (mkAnd patConcreteSet patFramedSet)
         )
 
+-- Given a function to scramble the arguments to concat, i.e.,
+-- @id@ or @reverse@, produces a pattern of the form
+-- `SetItem(absInt(X:Int)) Rest:Set`, or
+-- `Rest:Set SetItem(absInt(X:Int))`, respectively.
+selectFunctionPatternGen
+    :: Monad m
+    => ([CommonStepPattern Object] -> [CommonStepPattern Object])
+    -> PropertyT m (CommonStepPattern Object)
+selectFunctionPatternGen permutation = do
+    elementVar <- forAll (standaloneGen $ variableGen intSort)
+    frameVar <- forAll (standaloneGen $ variableGen setSort)
+    Monad.when (variableName elementVar == variableName frameVar) discard
+    let element = mkApp intSort absIntSymbol  [mkVar elementVar]
```

Comment:
You could reduce duplication between `selectPatternGen` and `selectFunctionPatternGen` by taking an argument which may (or may not) apply a symbol to the generated variable.